src/app/components/ATIAnalytics/params/index.test.ts

```diff
@@ -271,9 +271,28 @@ describe('ATIAnalytics params', () => {
         undefined,
         homePageAnalyticsData,
       );
-      expect(url).toMatchInlineSnapshot(
-        `"s=598285&s2=atiAnalyticsProducerId&p=kyrgyz.page&r=0x0x24x24&re=1024x768&hl=00-00-00&lng=en-US&x1=[urn%3Abbc%3Atipo%3Atopic%3Acm7682qz7v1t]&x2=[responsive]&x3=[atiAnalyticsAppName]&x4=[pcm]&x5=[http%253A%252F%252Flocalhost%252F]&x7=[index-home]&x8=[simorgh]&x9=[pageTitle]"`,
-      );
+
+      const parsedATIURLParams = new URLSearchParams(url as string);
+
+      const expectedATIURLParams = [
+        ['s', '598285'],
+        ['s2', 'atiAnalyticsProducerId'],
+        ['p', 'kyrgyz.page'],
+        ['r', '0x0x24x24'],
+        ['re', '1024x768'],
+        ['hl', '00-00-00'],
+        ['lng', 'en-US'],
+        ['x1', '[urn:bbc:tipo:topic:cm7682qz7v1t]'],
+        ['x2', '[responsive]'],
+        ['x3', '[atiAnalyticsAppName]'],
+        ['x4', '[pcm]'],
+        ['x5', '[http%3A%2F%2Flocalhost%2F]'],
+        ['x7', '[index-home]'],
+        ['x8', '[simorgh]'],
+        ['x9', '[pageTitle]'],
+      ];
```

Comment:
I've just seen your note on the other PR @pvaliani - what does a failing test look like if one of the params is incorrect? Is it easy to construct (the inline snapshot was super easy because it puts the data in for you) - here the data structure is not as easy to write (but 💯 easier to read).

If you're happy with this approach, do you plan to get the other tests working in a similar way? If it is too much hassle, you can leave it - I just wanted to stop and have a think before we make a lot of changes.

Reply:
Failures look like this (just replicated with typos):
<img width="874" alt="image" src="https://github.com/bbc/simorgh/assets/72317734/d16c2e33-f9bc-42b2-88ea-dc322e8c7898">

Creating the data structure initially to replace the other tests might be a bit tedious initially but it is for sure more readable. I think it would make sense to update all the tests in this file to use the same format, what do you think?
 We could introduce failures potentially (hopefully not) by getting rid of the snapshots and replacing with equality tests but those will at least be a bit more robust. I def think for this specific component it's helpful to make these tests be more readable. @andrewscfc seemed to agree too

Reply:
That test failure output is soooo much nicer than the snapshot one! And makes the code so much more readable too! You're right in that it could introduce failures, but we can pile in during code review and check it line by line, to make sure we haven't missed anything. Sometimes that's all we can do. 

Thanks for this, this will be really helpful in unravelling the spaghetti mess of this code. 🍝